### PR TITLE
Fixed the 'visible to Roles' in an API configuration issue

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
## Purpose
This fixes for the issue in the field of 'Visible to Roles' in an API configuration, is not acceptable for multiple roles with space after the comma separation.

## Goals
The access control logic should work for multiple roles even when there is a space after the comma.  (Eg: health-subscriber, education-subscriber)

https://github.com/wso2/product-apim/issues/3980

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java [hunk 1]
```diff
@@ -1239,7 +1240,8 @@ public static boolean jsFunction_addAPI(Context cx, Scriptable thisObj, Object[]
         }
 
         if (visibility != null && visibility.equals(APIConstants.API_RESTRICTED_VISIBILITY)) {
-            visibleRoles = (String) apiData.get("visibleRoles", apiData);
+            visibleRoles = ((String) apiData.get("visibleRoles", apiData));
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]:
```suggestion
            visibleRoles = (String) apiData.get("visibleRoles", apiData);
```
Remove the extra parentheses

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java [hunk 2]
```diff
@@ -1239,7 +1240,8 @@ public static boolean jsFunction_addAPI(Context cx, Scriptable thisObj, Object[]
...
+            visibleRoles = visibleRoles.replaceAll("\\s+","");
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: Can format both this and the other replaceAll line

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java [hunk 3]
```diff
@@ -920,6 +920,12 @@ public static boolean jsFunction_updateAPIDesign(Context cx, Scriptable thisObj,
 
         if (visibility != null && visibility.equals(APIConstants.API_RESTRICTED_VISIBILITY)) {
             visibleRoles = (String) apiData.get("visibleRoles", apiData);
+            StringBuilder roleBuilder = new StringBuilder();
+            for (String n : visibleRoles.split(",")) {
+                roleBuilder.append(n.trim()).append(",");
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: shall we rename n to role?

Comment (bhathiya) [reply components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: you can use String.join to join.

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java [hunk 4]
```diff
@@ -1215,6 +1221,7 @@ public static boolean jsFunction_addAPI(Context cx, Scriptable thisObj, Object[]
         String thumbUrl = (String) apiData.get("thumbUrl", apiData);
         String environments = (String) apiData.get("environments", apiData);
         String visibleRoles = "";
+        String visibleRolesArray[] ;
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: make this `String[] visibleRolesArray`

**Pull request merged**